waku/waku_lightpush/callbacks.nim

```diff
@@ -0,0 +1,66 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import
+ ../waku_core,
+ ../waku_relay,
+ ./common,
+ ./protocol,
+ ../waku_rln_relay,
+ ../waku_rln_relay/protocol_types,
+ ../common/ratelimit
+import
+ std/times,
+ libp2p/peerid,
+ stew/byteutils
+
+proc generateAndValidateRLNProof*(rlnPeer: Option[WakuRLNRelay], message: WakuMessage): (string, WakuMessage) =
+  var rlnResultInfo = ""
+  # TODO: check and validate if the message already has RLN proof?
+  if rlnPeer.isNone():
+    return (rlnResultInfo, message) # publishing message without RLN proof
+
+  # generate and append RLN proof
+  let
+    time = getTime().toUnix()
+    senderEpochTime = float64(time)
+  var msgWithProof = message
+  let appendProofRes = rlnPeer.get().appendRLNProof(msgWithProof, senderEpochTime)
+  if appendProofRes.isErr():
+    rlnResultInfo = "RLN proof generation failed: " & appendProofRes.error
+
+  return (rlnResultInfo, msgWithProof)
+
+proc getPushHandler*(
+    wakuRelay: WakuRelay,
+    rlnPeer: Option[WakuRLNRelay] = none[WakuRLNRelay]()
+): PushMessageHandler =
+  if wakuRelay.isNil():
+    debug "mounting lightpush without relay (nil)"
+    return proc(
+      peer: PeerId, pubsubTopic: string, message: WakuMessage
+    ): Future[WakuLightPushResult[void]] {.async.} =
+      return err("no waku relay found")
+  else:
+    debug "mounting lightpush with relay"
```

Comment:
Preferably these logs, related to mounting, should be in the `waku_node` not in the callbacks module. I'd move the logic of choosing which push handler to "get" to the `waku_node`: if relay is nil, log this case and `getNilPushHandler()`, else `getRelayPushHandler()`.